packages/firestore/src/remote/remote_store.ts

```diff
@@ -407,15 +426,43 @@ export class RemoteStore implements TargetMetadataProvider {
     }
 
     if (!snapshotVersion.isEqual(SnapshotVersion.min())) {
-      const lastRemoteSnapshotVersion = await this.localStore.getLastRemoteSnapshotVersion();
-      if (snapshotVersion.compareTo(lastRemoteSnapshotVersion) >= 0) {
-        // We have received a target change with a global snapshot if the snapshot
-        // version is not equal to SnapshotVersion.min().
-        await this.raiseWatchSnapshot(snapshotVersion);
+      try {
+        const lastRemoteSnapshotVersion = await this.localStore.getLastRemoteSnapshotVersion();
+        if (snapshotVersion.compareTo(lastRemoteSnapshotVersion) >= 0) {
+          // We have received a target change with a global snapshot if the snapshot
+          // version is not equal to SnapshotVersion.min().
+          await this.raiseWatchSnapshot(snapshotVersion);
+        }
+      } catch (e) {
+        logDebug(LOG_TAG, 'Failed to raise snapshot: ' + e);
```

Comment:
Is there any identifying information about the snapshot that we can log here?

Reply:
I am not sure if it will help, but I added the snapshot version.

Reply:
The snapshot version probably isn't that useful, but the target ID would be. Anything to help associate this message with its context.

If the targetId isn't available going back to what you had before is probably better.

Reply:
I changed this back to the original message. The target IDs are not easily available:

- DocumentWatchChange has `updatedTargetIds` and `removedTargetIds`
- ExistenceFilterChange has `targetId`
- WatchTargetChange has `targetIds`

If I understand the flow correctly, we should always be in a WatchTargetChange to raise a snapshot, but I don't want to bake this assumption into the code for better logging. I can invest some time here, but the targets are already in the Watch response that we log right before.